e2e/pageobjects/ide/OpenWorkspaceWidget.ts

```diff
@@ -0,0 +1,59 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { injectable, inject } from 'inversify';
+import { CLASSES } from '../../inversify.types';
+import { DriverHelper } from '../../utils/DriverHelper';
+import { By } from 'selenium-webdriver';
+
+@injectable()
+export class OpenWorkspaceWidget {
+    private static readonly OPEN_WORKSPACE_MAIN_VIEW_XPATH = '//div[@class=\'dialogTitle\']/div[text()=\'Open Workspace\']';
+    private static readonly OPEN_WORKSPACE_OPEN_BTN_CSS = 'div.dialogControl>button.main';
+    private static readonly THEIA_LOCATION_LIST_CSS = 'select.theia-LocationList';
+
+    constructor(@inject(CLASSES.DriverHelper) private readonly driverHelper: DriverHelper) {
+    }
+
+   async waitOpenWorkspaceWidget() {
```

Comment:
Here, and for all methods in this class. What do you think about customizing timeouts? Because for now you can just use default timeout.
Just do something like:
```
async waitOpenWorkspaceWidget(timeout: number = TestConstants.TS_SELENIUM_DEFAULT_TIMEOUT) {
```

and inside of the method:
```
        await this.driverHelper.waitVisibility(By.xpath(OpenWorkspaceWidget.OPEN_WORKSPACE_MAIN_VIEW_XPATH), timeout);
```

Reply:
Customized timeouts were applied